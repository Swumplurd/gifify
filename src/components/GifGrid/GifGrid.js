import {useEffect, useState} from "react";
import { GridItem } from "../GridItem/GridItem";

export const GifGrid = ({category}) => {

    const [gif, setGif] = useState([]);

    useEffect(() => {
        getGifs();
    }, []);

    const getGifs = async() => {
        const url = `https://api.giphy.com/v1/gifs/search?q=${category}&api_key=SSU4rxb9RPFUuGbxhXDXrS8VAcPmrDo4&limit=6`;
        const respuesta = await fetch(url);
        const {data} = await respuesta.json();

        const gifs = data.map((gif) => {
            return {
                id: gif.id,
                title: gif.title,
                url: gif.images?.downsized_medium.url,
                date: gif.import_datetime
            }
        })

        setGif(gifs)
    }

    return (
        <>
            <h3 className="text-center mt-5 text-uppercase">{category}</h3>
            <div className="row row-cols-1 row-cols-md-3 g-4">
                {
                    gif.map(item => {
                        return <GridItem key={item.id} {...item} />
                    })
                }
            </div>
        </>
    )
}