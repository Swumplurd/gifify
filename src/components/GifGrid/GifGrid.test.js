import { shallow } from "enzyme"
import { GifGrid } from "./GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs"
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en el componente GifGrid', () => {
    test('Debe hacer match con el snapshot', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const category = 'Morty'
        const wrapper = shallow(<GifGrid key={category} category={category}/>);
        expect(wrapper).toMatchSnapshot()
    })

    test('Debe de mostrar items cuando carga useFetchGifs', () => {
        const gifs = [{
            id: 123,
            title: 'Mory Malvado',
            url: 'https://www.img.gif',
            date: 121212
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const category = 'Morty'
        const wrapper = shallow(<GifGrid key={category} category={category}/>);
        
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GridItem').length).toBe(gifs.length)
    })
    
    
})
