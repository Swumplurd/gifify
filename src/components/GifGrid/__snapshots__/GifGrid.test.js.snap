// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGrid Debe de mostrar items cuando carga useFetchGifs 1`] = `
<Fragment>
  <h3
    className="text-center mt-5 text-uppercase"
  >
    Morty
  </h3>
  <div
    className="row row-cols-1 row-cols-md-3 g-4 animate__bounceIn"
  >
    <GridItem
      date={121212}
      id={123}
      key="123"
      title="Mory Malvado"
      url="https://www.img.gif"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el componente GifGrid Debe hacer match con el snapshot 1`] = `
<Fragment>
  <h3
    className="text-center mt-5 text-uppercase"
  >
    Morty
  </h3>
  <p
    className="text-center animate__heartBeat"
  >
    Cargando...
  </p>
  <div
    className="row row-cols-1 row-cols-md-3 g-4 animate__bounceIn"
  />
</Fragment>
`;
